### DISCUSSION TITLE
Updating a pretrained NER model

### DISCUSSION THREAD

**Discussion by lfcc1** [explosion/spaCy#8974] [Help: Coding & Implementations] [answered]
Hello,
Im trying to update an existing model but Im getting poor results.
Im trying 2 different approaches:

1 - Updating a spacy pre-trained model
2 - Updating a spacy custom ner model trained by myself.

Here is the code followed by the outputs
```
nlp = spacy.load("pt_core_news_lg") #  or custom model
doc = nlp("O Filipe Oliveira é o Conde da Barca")
print([(X.text,X.ent_iob_, X.ent_type_) for X in doc])

```
Outputs:

`
[('O', 'O', ''), ('Filipe', 'B', 'PER'), ('Oliveira', 'I', 'PER'), ('é', 'O', ''), ('o', 'O', ''), ('Conde', 'B', 'PER'), ('da', 'I', 'PER'), ('Barca', 'I', 'PER')]
`
As we can see, the model recognized some Portuguese entities.

Then, we have the model update with a single sentence:

```
line = ['O', 'Filipe', 'Oliveira', 'é', 'o', 'Conde', 'da', 'Barca']
labels = ['O', 'B-PER', 'I-PER', 'O', 'O', 'B-PER', 'O', 'B-PER']
doc = Doc(nlp.vocab, words=line)
example = Example.from_dict(doc, {"entities": labels})

for itn in range(100):
        nlp.update([example])

doc = nlp("O Filipe Oliveira é o Conde da Barca")
print([(X.text,X.ent_iob_, X.ent_type_) for X in doc])
```
Outputs:

`
[('O', 'O', ''), ('Filipe', 'O', ''), ('Oliveira', 'O', ''), ('é', 'O', ''), ('o', 'O', ''), ('Conde', 'O', ''), ('da', 'O', ''), ('Barca', 'O', '')]
`
We can see that the model was not able to recognize the entities and got even worst results than before the update.

As explained before, I also updated a custom pre-trained model and the results were similar to this (after the update, the model was unable to recognize entities it recognized before)

I´m I doing something wrong?

Thank you in advice

**Comment by svlandeg** [comment, marked as answer]
Hi!

Your approach in general is correct and you're right to expect overfitting when executing this.
Something sneaky is happening though:  your definition of `entities` does not follow the BILUO format, ultimately resulting in NO gold entities. You can check this by printing
```
print(example.reference.ents)
```
right after you've created the `example` -> you'll see that this is an empty list.

You need to change
```
labels = ['O', 'B-PER', 'I-PER', 'O', 'O', 'B-PER', 'O', 'B-PER']
```
to
```
labels = ['O', 'B-PER', 'L-PER', 'O', 'O', 'U-PER', 'O', 'U-PER']
```
then you'll see the gold entities
> (Filipe Oliveira, Conde, Barca)

and you'll see that now your code works as expected.

So in a nutshell: the input should be in BILUO format - not IOB. 

This is definitely confusing, but you can read the details of this format here: https://spacy.io/api/data-formats#dict-input
